stacks-signer/src/signerdb.rs

```diff
@@ -616,6 +640,15 @@ impl SignerDb {
         try_deserialize(result)
     }
 
+    /// Return the canonical tip -- the last globally accepted block.
+    pub fn get_canonical_tip(&self) -> Result<Option<BlockInfo>, DBError> {
```

Comment:
How do you resolve ties?  Ties happen in the wild all the time.

Reply:
We should never have ties in globally accepted blocks. We can have it in locally accepted blocks, but never globally.

Reply:
Sorry, your comment is unclear @jferrant. Are you saying that we should handle a tie or that we don't need to?

I think we may have the one case where there can be two globally accepted blocks at the same height -- at the tenure boundary if miner A is slow to mine the first block and the next bitcoin block arrives.

Reply:
I am agreeing with Jude. For the reason you mention, at the boundary, we could technically reorg the last block so we cannot have a canonical tip known without the context of a consensus hash.  

My comment is asking if we HAVE to know the canonical tip or if knowing just the last globally accepted block for a given consensus hash is enough (avoid the need for a tie breaker). If not, we will have to have a tie breaker as Jude mentions to accommodate the potential reorg at the boundary that you mention.

Reply:
Yeah I follow the issue.

I don't think we can take a consensus hash for this function, because the point of this function is to determine whether a block proposal's consensus hash is valid.

I do think using the last globally accepted block (by using `signed_group` as a tie breaker) works here, but I'm open to feedback. I'll add the `signed_group` tiebreaker for now, but happy to take a different approach.

Reply:
Addressed in [c8f1d99](https://github.com/stacks-network/stacks-core/pull/5452/commits/c8f1d999d6f6520b5a1a68ec237f215529351e48)